helix-rest/src/test/java/org/apache/helix/rest/server/TestResourceAssignmentOptimizer.java

```diff
@@ -0,0 +1,201 @@
+package org.apache.helix.rest.server;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import org.apache.helix.HelixDataAccessor;
+import org.apache.helix.TestHelper;
+import org.apache.helix.manager.zk.ZKHelixDataAccessor;
+import org.apache.helix.model.IdealState;
+import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+
+public class TestResourceAssignmentOptimizer extends AbstractTestClass {
+
+  String cluster = "TestCluster_2";
+  String instance1 = cluster + "dummyInstance_localhost_12930";
+  String swapNewInstance = "swapNewInstance";
+  String urlBase = "clusters/TestCluster_2/partitionAssignment/";
+  String swapOldInstance, toRemoveInstance;
+  HelixDataAccessor helixDataAccessor;
+  List<String> resources;
+  List<String> liveInstances;
+
+  @BeforeClass
+  public void beforeClass() {
+    helixDataAccessor = new ZKHelixDataAccessor(cluster, _baseAccessor);
+    _gSetupTool.addInstanceToCluster(cluster, instance1);
+    resources = _gSetupTool.getClusterManagementTool().getResourcesInCluster(cluster);
+    liveInstances =  helixDataAccessor.getChildNames(helixDataAccessor.keyBuilder().liveInstances());
+    Assert.assertFalse(resources.isEmpty() || liveInstances.isEmpty());
+
+    toRemoveInstance = liveInstances.get(0);
+    swapOldInstance = liveInstances.get(1);
+
```

Comment:
NIT: There're some blank lines in some files, and some typos in comments. Maybe you can double check and reformat.